Docs: improve "Environment Variables" page (#9256)
readthedocs/readthedocs.org: docs/user/environment-variables.rst

.. csv-table:: Environment Variables
   :header: Environment variable, Description, Example value
   :widths: 15, 10, 30

   ``READTHEDOCS``, Whether the build is running inside Read the Docs, ``True``
   ``READTHEDOCS_VERSION``, The Read the Docs slug of the version which is being built, ``latest``
   ``READTHEDOCS_VERSION_NAME``, Corresponding version name as displayed in Read the Docs' version switch menu, ``stable``
   ``READTHEDOCS_VERSION_TYPE``, Type of the event triggering the build, ``branch`` | ``tag`` | ``external`` (for :doc:`pull request builds </pull-requests>`) | ``unknown``
   ``READTHEDOCS_PROJECT``, The Read the Docs' slug of the project which is being built, ``my-example-project``
   ``READTHEDOCS_LANGUAGE``, The Read the Docs' language slug of the project which is being built, ``en``

.. note::

   The term slug is used to refer to a unique string across projects/versions containing ASCII characters only.
   This value is used in the URLs of your documentation.

User-defined environment variables
----------------------------------

If extra environment variables are needed in the build process (like an API token),

> you can define them from the project setting's page:

Reviewer: ```suggestion
you can define them from the project's settings page:
```

Environment Variables
=====================

Read the Docs uses some :ref:`default environment variables </builds:default-environment-variables>` when building your project.

> Besides, it provides a way to define your custom ones that will be merged with the default ones and used in the build process.

Author: I moved "Default environment variables" from "Build process" page to "Environment variables" page to introduce both types there and explain how to use them.

Besides, I added a link from the "Build process" to this page.

       'https://httpbin.org/basic-auth/username/password',
       auth=(username, password),
   )

> Also, you can read it from the :term:`user-defined build jobs` in the config file:

Reviewer: ```suggestion
You can also use any of these variables from :term:`user-defined build jobs` in your project's configuration file:
```
